e2e_test/js-tests/send_tx.mjs

```diff
@@ -1,44 +1,129 @@
+    },
+  },
 });
 
 const account = privateKeyToAccount(privateKey);
-const walletClient = createWalletClient({
-	account,
-	chain: devChain,
-	transport: http(),
-});
 
-const request = await walletClient.prepareTransactionRequest({
-	account,
-	to: "0x00000000000000000000000000000000DeaDBeef",
-	value: 2,
-	gas: 90000,
-	feeCurrency,
-	maxFeePerGas: 2000000000n,
-	maxPriorityFeePerGas: 0n,
+const publicClient = createPublicClient({
+  account,
+  chain: devChain,
+  transport: http(),
 });
-const signature = await walletClient.signTransaction(request);
-const hash = await walletClient.sendRawTransaction({
-	serializedTransaction: signature,
+const walletClient = createWalletClient({
+  account,
+  chain: devChain,
+  transport: http(),
 });
-console.log(hash);
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+async function waitBlocks(numBlocks) {
+  var initial = await publicClient.getBlockNumber({ cacheTime: 0 });
+  var next = initial;
+  while (next - initial < numBlocks) {
+    await sleep(500);
+    next = await publicClient.getBlockNumber({ cacheTime: 0 });
+  }
+}
+
+async function getTransactionReceipt(hash) {
+  try {
+    return await publicClient.getTransactionReceipt({ hash: hash });
+  } catch (e) {
+    if (e instanceof TransactionReceiptNotFoundError) {
+      return undefined;
+    }
+    throw e;
+  }
+}
+
+async function replaceTx(tx) {
+  const request = await walletClient.prepareTransactionRequest({
+    account: tx.account,
+    to: account.address,
+    value: 0n,
+    gas: 21000,
+    nonce: tx.nonce,
+    maxFeePerGas: tx.maxFeePerGas,
+    maxPriorityFeePerGas: tx.maxPriorityFeePerGas + 1000n,
+  });
+  const hash = await walletClient.sendRawTransaction({
+    serializedTransaction: await walletClient.signTransaction(request),
+  });
+  const receipt = await publicClient.waitForTransactionReceipt({
+    hash: hash,
+    confirmations: 1,
+  });
+  return receipt;
+}
+
+async function main() {
+  const request = await walletClient.prepareTransactionRequest({
+    account,
+    to: "0x00000000000000000000000000000000DeaDBeef",
+    value: 2n,
+    gas: 90000,
+    feeCurrency,
+    maxFeePerGas: 2000000000n,
+    maxPriorityFeePerGas: 0n,
+  });
+
+  var hash;
+
+  try {
+    hash = await walletClient.sendRawTransaction({
+      serializedTransaction: await walletClient.signTransaction(request),
+    });
+  } catch (e) {
+    // direct revert
+    console.log(
+      JSON.stringify({
+        success: false,
+        error: e,
+      }),
+    );
+    return;
+  }
+
+  var success = true;
+  // wait 1 second to give the node time to potentially process the tx
+  // in instamine mode. If the tx is invalidated, the block will not progress
+  await sleep(1000);
+  // give the node some time to process the transaction
+  // await waitBlocks(1);
+  var receipt = await getTransactionReceipt(hash);
+  if (!receipt) {
+    receipt = await replaceTx(request);
+    success = false;
+  }
```

Comment:
The problem with **solution 1**: if you don't have the original transaction available, it becomes unreliable to set correct gas-fee parameters in order to replace the transaction from the mempool, and in general it is more convenient to do it within the same script as the original send

**Solution 2** would be best, in order to separate the tests from each other in general. However this involves more setup and funds distribution, which I would like to avoid adding in this PR.

I added 2 more parameters to the `send_tx.mjs` script, so that you can explicitly control the behavior of how many blocks to wait for a transaction receipt, and wether to send a replace transaction if none is available after that time.